atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -0,0 +1,209 @@
+            elif option in ('disk_size'):
+                value = int(value)
+            opts[option] = value
+
+        section = 'ova-options'
+        if config.has_section(section):
+            ova = []
+            for k, v in config.items(section):
+                ova.append('{}={}'.format(k, v))
+            opts['ova_option'] = ova
+
+        section = 'factory-parameters'
+        if config.has_section(section):
+            factory = []
+            for option, value in config.items(section):
+                factory.append((option, value))
+            opts['factory_parameter'] = factory
+
+        # Set some defaults.
+        opts.setdefault('disk_size', 10)
+
+        return image_name, args, {'opts': opts}
+
+    def build_filesystem(self):
+        # Kickstart file should be in the same folder as Dockerfile
+        df_path, df_dir = self.workflow.source.get_dockerfile_path()
+        kickstart = os.path.join(df_dir, 'kickstart.conf')
+        if not os.path.exists(kickstart):
+            raise RuntimeError('Kickstart file not found: {}'
+                               .format(kickstart))
+
+        image_name, args, kwargs = self.parse_image_build_config(kickstart)
+        pattern = ('{}.*(\.tar|\.tar\.gz|\.tgz|\.bzip|\.tar\.xz|\.txz)$'
+                   .format(image_name))
+        filesystem_regex = re.compile(pattern, re.IGNORECASE)
+        if self.from_task_id:
+            task_id = self.from_task_id
+        else:
+            task_id = self.session.buildImageOz(*args, **kwargs)
+        return task_id, filesystem_regex
+
+    def find_filesystem(self, task_id, filesystem_regex):
+        for f in self.session.listTaskOutput(task_id):
+            f = f.strip()
+            match = filesystem_regex.match(f)
+            if match:
+                return task_id, match.group(0)
+
+        # Not found in this task, search sub tasks
+        for sub_task in self.session.getTaskChildren(task_id):
+            found = self.find_filesystem(sub_task['id'], filesystem_regex)
+            if found:
+                return found
+
+        return (None, None)
+
+    def download_filesystem(self, task_id, filesystem_regex):
+        task_id, file_name = self.find_filesystem(task_id, filesystem_regex)
+        if not task_id:
+            raise RuntimeError('Filesystem not found as task output: {}'
+                               .format(filesystem_regex.pattern))
+
+        self.log.info('Downloading filesystem: %s from task ID: %s',
+                      file_name, task_id)
+
+        contents = self.session.downloadTaskOutput(task_id, file_name)
+
+        self.log.info('Filesystem downloaded.')
+        return contents
+
+    def import_base_image(self, filesystem):
+        result = self.tasker.d.import_image_from_data(filesystem)
+        # Response not deserialized:
+        #   https://github.com/docker/docker-py/issues/1060
+        result = json.loads(result)
+        return result.get('status')
```

Comment:
If there is no `status` member we'll end up putting the string "None" in the FROM line. Maybe this should just be

```
return result['status']
```

so that we error out immediately?